parsing/parchmint-parser.js

```diff
@@ -59,6 +95,44 @@ class ParchmintParser {
 
         return layers;
     }
+
+    /**
+     * Parse a JSON object for the Component Features map.
+     *
+     * Fills the compFeatures map with Component Features with Component IDs as keys.
+     *
+     * @since 1.0.0
+     *
+     * @param {object}  jsonObj A parsed JSON object representing the Parchmint
+     *                          file.
+     */
+    parseComponentFeatures(jsonObj) {
+        jsonObj.features.forEach((value, index) => {
+            // Check whether this is a Component feature
+            if (value['x-span'] || value['y-span']) {
+               if (this.compFeatures.has(value.id)) {
+                   this.valid = false;
+                   console.log('Parser: Duplicate IDs (' + value.id + ') exist for the Component Features list.' +
+                           ' Skipping Component Feature with name "' + value.name + '" at index ' + index + '.');
+               } else {
+                   this.compFeatures.set(value.id, new ComponentFeature(value.name, value.layer, value['x-span'],
+                           value['y-span'], ParchmintParser.parseCoord(value.location), value.depth));
```

Comment:
Originally I was using the dot accessor, but because spans have dashes in the name I had to use brackets. I'll switch everything over to the bracket notation. Do you think I should switch everything in #53 and #54 to maintain consistency?

I don't think we need to validate that they exist in this function. I think the schema validation should check that for us.

Reply:
While I realize that JSON objects (once deserialized) are actual JavaScript objects I think it would be good to be consistent with the square bracket notation just to separate the JSON input from the internal objects (although I'm sure that's not idiomatic).

Reply:
Sounds good to me. I'll change them in the other PRs too.